Made major changes to Explore page and some minor to main and search page

Tushar-OP opened the pull request:
Changes made:
 
- Explore Page:
   1. Centered the text below poster in the expanded box for better look
   2. Changes the overall function of fetching of data from API
   3. Removed Future Builder, fixed the reload when scroll issue, added circular progress indictator while updating and cleaned up the code.

- main.dart:
   1. Made changes to avoid the bottom insets error on the search page.

agent515 commented:
TMDB API key was exposed in the api.dart file. I'm assuming it was done for testing purposes. I don't understand the removal of the Futurebuilder Widget on the Explore page. Any specific reasons?

Again confirming on the updates (other than creating apiKeyTMDB const):
updated packages and swapped Futurebuilder with Column

Tushar-OP commented:
> TMDB API key was exposed in the api.dart file. I'm assuming it was done for testing purposes. I don't understand the removal of the Futurebuilder Widget on the Explore page. Any specific reasons?

Yeah the key was exposed in the api.dart so I used it for testing, you should remove the commented part also it has the key. Also the future builder was giving a issue while handling updates; to be precise while getting a new page. So I made changes to fix that behaviour and found out that it can be done without future builder so I removed it's implementation.

> Again confirming on the updates (other than creating apiKeyTMDB const):
> updated packages and swapped Futurebuilder with Column

Yeah, the packages were updated by flutter itself I didn't meddle with it, and removed Futurebuilder
